core/src/main/java/org/opensearch/sql/planner/physical/PaginateOperator.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.planner.physical;
+
+import java.util.List;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.executor.ExecutionEngine;
+import org.opensearch.sql.planner.physical.PhysicalPlan;
+import org.opensearch.sql.planner.physical.PhysicalPlanNodeVisitor;
+import org.opensearch.sql.planner.physical.ProjectOperator;
+
+@RequiredArgsConstructor
+@EqualsAndHashCode(callSuper = false)
+public class PaginateOperator extends PhysicalPlan {
```

Comment:
Could you explain what this used for? Is it in-memory paginate implementation for unsupported query or other storage engine?

Reply:
Exactly -- an in-memory paginate implementation for cases where it cannot be pushed down.

Reply:
I didn't find any case of in-memory pagination. Could you give an example?
I tried to get rid of `PaginateOperator` in [#1499](https://github.com/opensearch-project/sql/pull/1499#discussion_r1159021420). I also don't serialize it #1498, because it is not needed, once `PushPageSize` did its work.

Reply:
@Yury-Fridlyand yes, that's also my question. I thought after pushdown, there should be no operator for paginate in query plan tree.

Reply:
Once #1498 is merged, we can remove PaginateOperator from the execution plan during optimization.

Reply:
WIP in #1528

Reply:
Removed